packages/plugingen/src/commands/generate/phases/plugin-actions/management-actions.ts

```diff
@@ -0,0 +1,124 @@
+  const { addImport, pluginConfig, addType } = input;
+  if (pluginConfig.installConfig != null) {
+    addImports(
+      addImport,
+      pluginConfig.installConfig.dependencies?.map((x) => x.plugin) ?? []
+    );
+
+    const initArgs = pluginConfig.installConfig.initAbiParams ?? [];
+
+    addType("InstallArgs", JSON.stringify(initArgs));
+    addType(
+      `Install${pluginConfig.name}Params`,
+      dedent`{
+        args: Parameters<typeof encodeAbiParameters<InstallArgs>>[1];
+        pluginAddress?: Address;
+        dependencyOverrides?: FunctionReference[];
+    }`,
+      true
+    );
+    addType(
+      "ManagementActions<TAccount extends SmartContractAccount | undefined = SmartContractAccount | undefined>",
+      dedent`{
+      install${pluginConfig.name}: (args: {overrides?: UserOperationOverrides} & Install${pluginConfig.name}Params & GetAccountParameter<TAccount>) => Promise<SendUserOperationResult>
+    }`
+    );
+
+    const dependencies = (pluginConfig.installConfig.dependencies ?? []).map(
+      (x) => dedent`
+        (() => {
+          const pluginAddress = ${x.plugin.name}.meta.addresses[chain.id];
+          if (!pluginAddress) {
+            throw new Error("missing ${x.plugin.name} address for chain " + chain.name);
+          }
+
+          return encodePacked(
+            ["address", "uint8"],
+            [pluginAddress, ${x.functionId}]
+          );
+        })()
+      `
+    );
+
+    const installMethodName = `install${pluginConfig.name}`;
+
+    input.content.push(dedent`
+    ${installMethodName}({account = client.account, overrides, ...params}) {
+      if (!account) {
+        throw new AccountNotFoundError();
+      }
+
+      if (!isSmartAccountClient(client)) {
+        throw new IncompatibleClientError("SmartAccountClient", "${installMethodName}", client);
+      }
+
+      const chain = client.chain;
+      if (!chain) {
+        throw new ChainNotFoundError();
+      }
+
+      const dependencies = params.dependencyOverrides ?? [${dependencies.join(
+        ",\n\n"
+      )}];
+      const pluginAddress = params.pluginAddress ?? ${
+        pluginConfig.name
+      }.meta.addresses[chain.id] as Address | undefined;
+
+      if (!pluginAddress) {
+        throw new Error("missing ${
+          pluginConfig.name
+        } address for chain " + chain.name);
+      }
+
+      return installPlugin_(client, {
+        pluginAddress,
+        pluginInitData: encodeAbiParameters(${JSON.stringify(
+          initArgs
+        )}, params.args),
+        dependencies,
+        overrides,
+        account,
+      });
+    }
+  `);
+  }
+
+  return input;
+};
+
+const addImports = (
+  addImport: PhaseInput["addImport"],
+  deps?: PluginConfig[]
+) => {
+  if (deps != null && deps.length > 0) {
+    addImport("viem", { name: "encodePacked" });
+    deps.forEach((x) => {
+      addImport(
+        `../${kebabCase(x.name.replaceAll(/[pP]lugin/g, ""))}/plugin.js`,
```

Comment:
wait no... not here this is meant to import the dependent plugin. this piece is correct

Reply:
sorry I meant aa-alchemy, not aa-accounts and we move MultiOwnerPlugin, sessionkeyplugin to aa-alchemy (on this pr?). 

plugingen is meant to be a standalone package that is used as "devDependency" no? similar to wagmi-cli, think plugingen-cli is more suitable to be a devDependency for projects to create typings for plugins they develop. If so we should not have any relative path imports for generated plugin code because the projects that use/integrate plugingen will not be able to resolve the relative path imports. 

I think const `GenerateSchema` should not only include path to plugingen config file, and the directory to search for the config file, but also the "output path". Also, for PluginGenConfig, the `installConfig` field should include the import path of the PluginGenConfig of the plugin that it depends on.

Reply:
no this should be referencing `aa-alchemy` either. This is generated based on the `deps` in the install config. You're right it can come from anywhere, but this assumes that developers building plugins and thus generating code for them are, for now, also building their dependencies if there are any. That assumption I think is ok for now and results in this relative import because the dep will be generated alongside the plugin. 

If we want to make the import path for deps be overrideable so that it can be pulled in from other packages, we should do that in a separate PR. this is just meant to be the bare minimum PR to cover the existing tool but as standalone CLI

Reply:
also we're not going to be moving Plugins or accounts in this PR. We'll be moving things over to account-kit first and then deprecating them in a later PR. This PR is JUST the plugingen breakout into its own CLI tool

Reply:
created an issue to track this extension here: https://github.com/alchemyplatform/aa-sdk/issues/511